src/builder.cr

```diff
@@ -128,9 +128,13 @@ module Mint
       end
     end
 
-    def index(css_prefix, relative, optimize)
+    def index(css_prefix, relative, optimize, runtime_path)
       runtime =
-        Assets.read("runtime.js")
+        if runtime_path
```

Comment:
Maybe we should check if the specified file exists? This way it will raise an exception if the file is not found - ditto on the compile command as well.

It's not blocking this PR just would be nice to have :slightly_smiling_face:

Reply:
What type of exception would you expect to be thrown? Also, If I'm adding a bit for complex logic, would it make sense to extract this to a module, so I don't duplicate it ?

Reply:
`File::NotFoundError` should be thrown if the file doesn't exist by the `File.read`. Should we use something like `print` to stderr and exit with 1, rather than having crystal print all stack trace with it?

Reply:
> Should we use something like print to stderr and exit with 1, rather than having crystal print all stack trace with it?

Yeah, that's what I was thinking of.

> Also, If I'm adding a bit for complex logic, would it make sense to extract this to a module, so I don't duplicate it ?

Ideally yes, but I'm not sure two occurrences warrants it or not, @Sija what do you think?

Reply:
> Ideally yes, but I'm not sure two occurrences warrants it or not, @Sija what do you think?

I don't think so, nope. It gives almost no gain considering it saves a few lines, and creates additional abstraction and a dependency.

Reply:
> What type of exception would you expect to be thrown?

@WhoAteDaCake I'd suggest creating sth like `RuntimeFileNotFound` `message` (akin to `EnvFileNotFound`) and register it as a `command_error`.

Reply:
I've released that we're also using `runtime.js` within `sandbox_server.cr` and `test_runner.cr`. Should both of those also allow for swapping the runtime? If so, could you provide me a bit of guidance for the approach ?

Reply:
I'd think so. /cc @gdotdesign